api/src/main/java/org/pmiops/workbench/workspaces/WorkspaceServiceImpl.java

```diff
@@ -366,12 +388,38 @@ public Workspace findByWorkspaceId(long workspaceId) {
       Map<String, WorkspaceAccessEntry> rolesMap) {
     List<UserRole> userRoles = new ArrayList<>();
     for (Map.Entry<String, WorkspaceAccessEntry> entry : rolesMap.entrySet()) {
+      // Filter out groups
       User user = userDao.findUserByEmail(entry.getKey());
-      userRoles.add(workspaceMapper.toApiUserRole(user, entry.getValue()));
+      if (user == null) {
+        log.log(Level.WARNING, "No user found for " + entry.getKey());
+      } else {
+        userRoles.add(workspaceMapper.toApiUserRole(user, entry.getValue()));
+      }
     }
     return userRoles.stream()
         .sorted(
             Comparator.comparing(UserRole::getRole).thenComparing(UserRole::getEmail).reversed())
         .collect(Collectors.toList());
   }
+
+  @Override
+  public Workspace setPublished(
+      Workspace workspace, String publishedWorkspaceGroup, boolean publish) {
+    Map<String, WorkspaceAccessEntry> firecloudAcls =
+        getFirecloudWorkspaceAcls(workspace.getWorkspaceNamespace(), workspace.getFirecloudName());
+
+    Map<String, WorkspaceAccessLevel> updatedAcls = new HashMap<>();
+    for (Map.Entry<String, WorkspaceAccessEntry> entry : firecloudAcls.entrySet()) {
+      updatedAcls.put(
+          entry.getKey(), WorkspaceAccessLevel.fromValue(entry.getValue().getAccessLevel()));
+    }
+
+    if (publish) {
+      updatedAcls.put(publishedWorkspaceGroup, WorkspaceAccessLevel.READER);
+    } else {
+      updatedAcls.put(publishedWorkspaceGroup, WorkspaceAccessLevel.NO_ACCESS);
+    }
+
+    return updateWorkspaceAcls(workspace, updatedAcls);
```

Comment:
This feels different enough from `updateWorkspaceAcls` that I would just call `fireCloudService.updateWorkspaceACL` directly with a single input. Most of what that method doing is not necessary for what you're trying to achieve here.